reark/src/main/java/io/reark/reark/data/stores/SingleItemContentProviderStore.java

```diff
@@ -52,11 +53,12 @@
  * @param <T> Type of the data this store contains.
  * @param <U> Type of the id used in this store.
  */
-public abstract class SingleItemContentProviderStore<T, U> extends ContentProviderStore<T> {
+public abstract class SingleItemContentProviderStore<T, U> extends ContentProviderStoreCoreBase<T>
```

Comment:
We could name this just ContentProviderStore, as we don't have other kinds of content provider stores available.

Also the put() needs @Override